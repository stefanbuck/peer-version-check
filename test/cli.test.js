const fs = require('fs');
const path = require('path');
const cp = require('child_process');

const cli = path.join(__dirname, '../cli.js');
const fixtures = path.join(__dirname, 'fixtures');

function run(file) {
  console.log(file);
  return new Promise(resolve => {
    cp.execFile(cli, [file], (error, stdout) => {
      resolve({
        exitCode: (error && error.code) || 0,
        stdout: stdout.toString()
      });
    });
  });
}

const fixutres = fs.readdirSync(fixtures);
const jsonFiles = fixutres
  .filter(name => path.extname(name) === '.json');

jsonFiles.forEach(file => {
  const name = file.replace(/_/g, ' ').replace('.json', '');
  it(name, async () => {
    const out = await run(path.join(fixtures, file));
    expect(out).toMatchSnapshot();
  });
});
