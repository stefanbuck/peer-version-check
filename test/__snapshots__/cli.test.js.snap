// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`good 1`] = `
Object {
  "exitCode": 0,
  "stdout": "",
}
`;

exports[`missing devDependency 1`] = `
Object {
  "exitCode": 1,
  "stdout": "Expected devDependency 'a' to be:
   ^1.0.0
Found:
   undefined

Expected devDependency 'b' to be:
   ^1.0.0
Found:
   undefined

",
}
`;

exports[`missing peerDependency 1`] = `
Object {
  "exitCode": 1,
  "stdout": "Expected devDependency 'b' to be:
   ^1.0.0
Found:
   undefined

",
}
`;

exports[`no peerDependency 1`] = `
Object {
  "exitCode": 0,
  "stdout": "",
}
`;
